ytypes/string_type.go

```diff
@@ -26,6 +26,35 @@ import (
 
 // Refer to: https://tools.ietf.org/html/rfc6020#section-9.4.
 
+var (
+	POSIXregexCache = map[string]*regexp.Regexp{}
+	RE2regexCache   = map[string]*regexp.Regexp{}
+)
+
+// compilePattern returns the compiled regex for the given regex
+// pattern. It caches previous look-ups for faster performance.
+// Go's regexp implementation might be relatively slow compared to other
+// languages: https://github.com/golang/go/issues/11646
+func compilePattern(pattern string, isPOSIX bool) (*regexp.Regexp, error) {
```

Comment:
Is this function adequately tested from the unit tests that we have today? I presume so, but just wanted you to take a look and confirm.

Reply:
Yes I believe so. The current tests for `ValidateStringRestrictions` verifies many regexes, and some of the tests use the same regex, which tests the caching code. Although there isn't a mock to verify the caching behaviour.